ci: update mergify rules to include teams

Madhu-1 opened the pull request:
updated mergify rules to consider the team's approval to merge a PR.

more details at #2367

fixes #2367

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: update mergify rules to include teams

updated mergify rules to consider the teams
approval to merge a PR.

more details at #2367

fixes #2367

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
CC @ceph/ceph-csi-contributors @ceph/ceph-csi-maintainers

nixpanic commented on .mergify.yml:
```diff
@@ -36,8 +36,9 @@ pull_request_rules:
       - base~=^(devel)|(release-.+)$
-      - "#approved-reviews-by>=2"
       - "#changes-requested-reviews-by=0"
+      - "approved-reviews-by=@ceph/ceph-csi-contributors"
```
I wonder if the notation should be `@ceph-csi-contributors` without the `ceph/` organization part. Did you find an example configuration file where teams are used?

Madhu-1 replied:
took example from https://github.com/Mergifyio/mergify-engine/issues/310#issuecomment-444419127

Rakshith-R replied:
I guess @ceph/ceph-csi-contributors  is working 
https://dashboard.mergify.io/github/ceph/config-editor?repositoryName=ceph-csi&pullRequestNumber=2425

Madhu-1 commented:
This needs a manual merge. @humblec @nixpanic feel free to merge this PR after approval.

Rakshith-R commented:
@Madhu-1 , we need to preserve `- "#approved-reviews-by>=2"`
since a maintainer is also a contributor, its being counted twice, details in the link

https://dashboard.mergify.io/github/ceph/config-editor?repositoryName=ceph-csi&pullRequestNumber=2423

Madhu-1 commented:
> @Madhu-1 , we need to preserve `- "#approved-reviews-by>=2"`
> since a maintainer is also a contributor, its being counted twice, details in the link
> 
> https://dashboard.mergify.io/github/ceph/config-editor?repositoryName=ceph-csi&pullRequestNumber=2423

Yes correct. updating it

Rakshith-R commented on .mergify.yml:
```diff
@@ -122,7 +124,9 @@ pull_request_rules:
       - label!=DNM
       - "#changes-requested-reviews-by=0"
-      - "#approved-reviews-by>=1"
+      - "#approved-reviews-by>=2"
+      - "approved-reviews-by=@ceph/ceph-csi-contributors"
+      - "approved-reviews-by=@ceph/ceph-csi-maintainers"
```
```suggestion
      - "#approved-reviews-by>=2"
      - "approved-reviews-by=@ceph/ceph-csi-contributors"
      - "approved-reviews-by=@ceph/ceph-csi-maintainers"
```

Backport prs also need two approvals now ?

Madhu-1 replied:
Yes as we have more people let's have 2 approvals

humblec commented:
@Madhu-1 i think its good to update the contribution guide with this too?

Rakshith-R reviewed (commented):
Same changes missing for backport release v1.2 branch (line 105/107)but maybe remove backport pr rules for v1.2 completely too? 
Otherwise LGTM.

Madhu-1 committed:
doc: update development guide for new rules

updated development guide requirement to
have review from contributors and reviewers.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
> @Madhu-1 i think its good to update the contribution guide with this too?

updated @humblec PTAL

Madhu-1 commented:
> Same changes missing for backport release v1.2 branch (line 105/107)but maybe remove backport pr rules for v1.2 completely too?
> Otherwise LGTM.

v1.x is getting deprecated will remove all mergify rules for that in as a follow-up PR.

humblec commented:
@nixpanic are you good with the changes here?

Madhu-1 commented:
Merging the PR as it has required approvals.

Madhu-1 commented:
> Sorry but I didn't understand the command. Please consult [the commands documentation](https://docs.mergify.io/commands.html) .
> 
> _Hey, I reacted but my real name is @Mergifyio_

issue already reported for this at https://github.com/Mergifyio/mergify-engine/issues/3068
